nautobot/docs/user-guide/administration/upgrading/from-v1/ipam-upgrade-guide.md

```diff
@@ -0,0 +1,191 @@
+    - A`VRF` must be assigned to a `Device` before it may be assigned to an `Interface`
+    - A`VRF` must be assigned to a `VirtualMachine` before it may be assigned to an `VMInterface`
+- This addresses a fundamental flaw in which an Interface could have multiple IPAddresses assigned with conflicting VRFs, which is impossible in practice.
+
+### Primary IPv4/IPv6 no longer unique
+
+- [#3939](https://github.com/nautobot/nautobot/issues/3939) - Changed `Device.primary_ip4` and `primary_ip6` fields from `OneToOneField`to `ForeignKey`, relaxing the uniqueness constraint.
+- [#3939](https://github.com/nautobot/nautobot/issues/3939) - Changed `VirtualMachine.primary_ip4` and `primary_ip6` fields from `OneToOneField` to `ForeignKey`, relaxing the uniqueness constraint.
+- Because now same IP used in multiple places requires this.
+
+## Preparing your IPAM Data for Nautobot 2.0
+
+- IPaddress VRF assignments must be unique
+- Parents and IPAddresses must share the same Namespace
+- IPAddress objects can no longer be orphaned
+    - The database migration will automatically create a parent prefix for IPAddresses that do not have an eligible parent. For example 1.2.3.4/32 will have a parent Prefix created of the same network and prefix_length e.g. 1.2.3.4/32.
+    - If you do not wish for these single host prefixes to be created, create a parent prefix of your desired size to contain them before upgrading
+- You cannot migrate Interfaces or VMInterfaces that have IPs with differing VRFs
+    - Make sure for interfaces with multiple IPs that each IP assigned to the same Interface share the same VRF or are not assigned a VRF
+
+## After you Upgrade
+
+This section includes various things to consider after you have successfully upgraded Nautobot.
+
+### Cleanup Namespaces
+
+> This also applies to any "VRF Namespace" objects that were created.
+
+A priority of the upgrade process is to assert that no data will be lost. Due to the introduction of strict uniqueness constraints to disallow duplicate Prefix, IPAddress, and VRF objects within the same Namespace.
+
+- If you need to swap an object into another Namespace (say Global and Cleanup Namespace 1)
+    - Create a third "Temporary" namespace
+    - Move objects from Global to Temporary
+    - Move objects from Cleanup Namespace 1 to Global
+    - Move objects from Temporary to Cleanup Namespace 1
+    - Delete Temporary
+- Tenant affinity
+    - A best effort is made to keep Prefixes and IPAddresses together by shared attributes such as tenant, but this is not alway possible for various reasons such as numerous duplicate with identical or too-closely-similar criteria.
+
+### Delete duplicate objects
+
+- No data loss is prioritized, therefore some objects that may have been duplicates before, may no longer be needed. Some examples include:
+    - The same IP assigned to multple interfaces. Where possible, a single IP is now assigned
+    - VRFs that were used strictly for custom uniqueness boundaries
```

Comment:
Suggest adding a note/cross-reference here to the docs about the IPAddress merge tool.